### PR TITLE
Use teaser element for contact boxes.

### PR DESCRIPTION
**Pull request by CarstenHoyer**
Screenshot 2020-06-11 at 14 05 06

The input can contain inline css, that cannot be cleansed out, so I need to force overwrite it.

KKB-348

### CODE REVIEW

File: kkb_contact/css/kkb_contact.css [hunk 1]
```diff
@@ -46,12 +48,27 @@
           box-shadow 300ms cubic-bezier(0.165, 0.84, 0.44, 1);
 }
 
+.kkb-contact-teaser-content .body {
+  color: #464646;
+}
+
+/** Force children to have these styles,
+ *  because the variable we inject has its own style.
+ *  Probably because it was pasted from Word.
```
Comment (xendk) [inline kkb_contact/css/kkb_contact.css:57]: It shouldn't. Either the input filter should strip it, or they should use the "paste from word" plugin, copy it to Notepad first, or just stop using Word. "Fixing" it in CSS is not the right solution.

File: kkb_contact/kkb_contact.pages.inc [hunk 2]
```diff
@@ -54,23 +54,16 @@ function kkb_contact_page() {
   $vars['write_section_title'] = variable_get('kkb_contact_page_write_section_title', KKB_CONTACT_PAGE_PHONE_SECTION_TITLE_DEFAULT);
 
   foreach (['first', 'second'] as $box) {
-    $vars['write_' . $box . '_title'] = variable_get('kkb_contact_page_write_' . $box . '_title', constant('KKB_CONTACT_PAGE_WRITE_' . strtoupper($box) . '_TITLE_DEFAULT'));
-
-    $body = variable_get('kkb_contact_page_write_' . $box . '_body',
+    $box_content['title'] = variable_get('kkb_contact_page_write_' . $box . '_title', constant('KKB_CONTACT_PAGE_WRITE_' . strtoupper($box) . '_TITLE_DEFAULT'));
+    $box_content['body'] = variable_get('kkb_contact_page_write_' . $box . '_body',
```
Comment (xendk) [inline kkb_contact/kkb_contact.pages.inc:58]: Why are you removing the `check_markup` call which applies the input filter? You're removing the XSS protection and breaking using other formats.

File: kkb_contact/kkb_contact.theme.inc [hunk 3]
```diff
@@ -15,19 +15,27 @@ function template_preprocess_kkb_contact_teaser(&$variables) {
...
   }
-  else {
-    $variables['title'] = $teaser['title'];
-    $variables['link'] = '';
+//  else {
+//
+//    $variables['link'] = '';
+//  }
+
+  if (array_key_exists('value', $teaser['body'])) {
```
Comment (xendk) [inline kkb_contact/kkb_contact.theme.inc:34]: As you removed `check_markup`, `value` is always set, which means that it'll always use `$teaser['body']['value']` which contains unsanitized data.

You're resusing `kkb_contact_teaser` for something it wasn't meant for, as regular teasers has plain text as body, while the contact page boxes has WYSIWYG as body.

If you really want to do that, it'll have to support both options properly:

Defining `$teaser['body']` as "either a string or a input filter array" is bad interfacing.
You can't let `kkb_contact_page` do the `check_markup` and pass that as body, as `check_plain` will munge the HTML.
Defining it as "sanitized HTML", means you'll have to find every place that calls `theme('kkb_contact_teaser')` and ensure that they `check_plain` the body, and you don't want to do that.

So the least painful solution is to have `kkb_contact_page` do `check_markup` on its input filter data and pass it in as `$teaser['body_formatted']`, and have `template_preprocess_kkb_contact_teaser` use that, verbatim, if set or fall back to `check_plain`ing `$teaser['body']`.

File: kkb_contact/templates/kkb-contact-page.tpl.php [hunk 4]
```diff
@@ -38,19 +38,8 @@
...
-      </div>
-      <?php print $write_first_link ?>
-    </div>
-    <div class="kkb-contact-write kkb-contact-write-second kkb-contact-box">
-      <h3><?php print $write_second_title ?></h3>
-      <div class="body">
-        <?php print $write_second_body ?>
-      </div>
-      <?php print $write_second_link ?>
-    </div>
```
Comment (xendk) [inline kkb_contact/templates/kkb-contact-page.tpl.php:54]: Shouldn't here be some matching CSS deleted?

Comment (CarstenHoyer) [reply kkb_contact/templates/kkb-contact-page.tpl.php:54]: Surprisingly little, but there was a little bit.

File: kkb_contact/kkb_contact.theme.inc (hunk 3 above)
Comment (CarstenHoyer) [reply kkb_contact/kkb_contact.theme.inc]: * I re-introduced check_markup
* I made special theme for these slightly different teasers (I called them boxes for lack of a better word)

File: kkb_contact/kkb_contact.pages.inc (hunk 2 above)
Comment (CarstenHoyer) [reply kkb_contact/kkb_contact.pages.inc:58]: See other comment

**Pull request merged**